### ISSUE TITLE
Option Reading Problem?

### ISSUE THREAD

**Issue by rpspringuel** [gregorio-project/gregorio-test#4]
When running the test scrip I get the following output at the beginning of the run:

```
./gregorio-test.sh: line 63: declare: -A: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
```

This appears to be occurring both with and without options (in particular I tested using the `-S` option).

**Comment by rpspringuel** [comment]
Oh, the script runs fine after it outputs this message.

**Comment by henryso** [comment]
I wonder what version of bash you're using.  The `-A` option is a bash 4 feature.

**Comment by rpspringuel** [comment]
Checking:

```
GNU bash, version 4.3.39(1)-release (x86_64-apple-darwin14.3.0)
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

**Comment by henryso** [comment]
Then it's a bit odd.  Maybe the script is calling a bash 3 somewhere on your system?  Or the declare built-in is somehow the older version?

**Comment by rpspringuel** [comment]
I looked at `gregorio-test.sh` and I noticed that you explicitly invoked `/bin/bash`.  Checking that I get:

```
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin14)
Copyright (C) 2007 Free Software Foundation, Inc
```

So you are correct in supposing that a different version of bash is being invoked.  I get the updated bash through MacPorts, so it's in `/opt/local/bin/bash`, not `/bin/bash`.  Changing the first line of the script to `#!/usr/bin/env bash` fixes this problem.  Will that change create problems for anyone else?
